src/arm64/dom_parser_implementation.cpp

```diff
@@ -105,7 +105,10 @@ WARN_UNUSED error_code dom_parser_implementation::stage1(const uint8_t *_buf, si
   this->len = _len;
   return arm64::stage1::json_structural_indexer::index<64>(buf, len, *this, streaming);
 }
-
+#include "generic/stage1/utf8_validator.h"
+WARN_UNUSED bool implementation::validate_utf8(const char *buf, size_t len) const noexcept {
+  return simdjson::arm64::stage1::utf8_validate(buf,len);
```

Comment:
Can we name both of them validate_utf8? No reason to have two names, seems like.

Reply:
It is deliberate on my part because I spent about 2 hours trying to debug this damn code to understand why, for example, we have about 5 different functions called minify, all with small differences, and all under the same damn name. I find it confusing to have two functions called validate_utf8 in the same mental scope. I want to think of them as being different.

I will rename so that my intentions are clearer.

Reply:
Ok. I renamed the generic function "generic_validate_utf8". My intention is that when the reader arrives at this function, they can just search the name to find its definition. Having all functions named the same, only differentiated by the namespace... it makes it hard (for me and maybe others) to understand. When I look at the code... I want to know right away what function is defined where.